usb/device/targets/TARGET_STM/USBPhy_STM32.cpp

```diff
@@ -213,39 +213,34 @@ void USBPhyHw::init(USBPhyEvents *events)
     defined(TARGET_DISCO_F413ZH) || \
     defined(TARGET_DISCO_F469NI) || \
     defined(TARGET_DISCO_F746NG_OTG_FS)
-    __HAL_RCC_GPIOA_CLK_ENABLE();
     pin_function(PA_11, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_NOPULL, GPIO_AF10_OTG_FS)); // DM
     pin_function(PA_12, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_NOPULL, GPIO_AF10_OTG_FS)); // DP
+#if defined(TARGET_DISCO_F746NG_OTG_FS)
+    pin_function(PJ_12, STM_PIN_DATA(STM_MODE_INPUT, GPIO_NOPULL, GPIO_AF10_OTG_FS));  // VBUS
+#else
     pin_function(PA_9, STM_PIN_DATA(STM_MODE_INPUT, GPIO_NOPULL, GPIO_AF10_OTG_FS));  // VBUS
+#endif
     pin_function(PA_10, STM_PIN_DATA(STM_MODE_AF_OD, GPIO_PULLUP, GPIO_AF10_OTG_FS)); // ID
     pin_function(PA_8, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_NOPULL, GPIO_AF10_OTG_FS));  // SOF
     __HAL_RCC_USB_OTG_FS_CLK_ENABLE();
 
 #elif defined(TARGET_DISCO_F429ZI)
-    __HAL_RCC_GPIOB_CLK_ENABLE();
```

Comment:
This is not related to F7 ?

Reply:
No this is common to all STM32. This is why I put STM32 instead of STM32F7 in the commit title. I can send this commit in a separate PR for clarity. @0xc0170 your opinion ?

Reply:
no, this means you should test again F429 ?

Reply:
No tested ok on one F7 board that's enough.

Reply:
Commit removed. Discussion closed...